control/config.py

```diff
@@ -20,7 +20,40 @@
     'control.squeeze_time_response': None,
     'forced_response.return_x': False,
 }
-defaults = dict(_control_defaults)
+
+
+class DefaultDict(dict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __setitem__(self, key, value):
+        super().__setitem__(self._check_deprecation(key), value)
+
+    def __missing__(self, key):
+        repl = self._check_deprecation(key)
+        if self.__contains__(repl):
+            return self[repl]
+        else:
+            raise KeyError
+
+    def copy(self):
+        return DefaultDict(self)
+
+    def get(self, key, default=None):
+        return super().get(self._check_deprecation(key), default)
```

Comment:
Overriding `get` but not `__getitem__` ?

Reply:
`__getitem__` calls `__missing__` if the key is not present. So the check and warning message are within the code path for `x = config.defaults['module.oldkey`]